src/main/java/com/ably/kafka/connect/ChannelConfig.java

```diff
@@ -0,0 +1,76 @@
+package com.ably.kafka.connect;
+
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.ChannelMode;
+import io.ably.lib.types.ChannelOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CHANNEL_CONFIG;
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CLIENT_CHANNEL_CIPHER_KEY;
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CLIENT_CHANNEL_PARAMS;
+
+interface ChannelConfig {
+    String getName();
+
+    ChannelOptions getOptions() throws ChannelSinkConnectorConfig.ConfigException;
+}
+
+class ChannelConfigImpl implements ChannelConfig {
```

Comment:
For what it's worth, I've always considered the Java convention of an `Impl` (abbreviation) suffix on class names pretty opaque and non-inclusive for many maintainers, especially those not deeply familiar with some of the more ancient idioms of Java. I also have a preference for using prefixes for subclasses / implementers, rather than suffixes - so, for me, I would consider renaming this class to `DefaultChannelConfig`.

I note you already have a `DefaultChannelSinkMapping.java` file, containing a `DefaultChannelSinkMapping` class implementing `ChannelSinkMapping`, so perhaps the convention I prefer is already established in this repository?

Finally, perhaps consider moving this class into its own file. I think we should be sticking to a single interface or class per file, matching the file name.

Reply:
That's really a good point and to be honest, I do not like -Impl postfix for class names either. I think I was recently exposed to some code bases that uses this convention that clouded and corrupted my vision 😄  Usually when there is a single class and there is a need to provide a fake implementation for tests, this is used very commonly, probably by old Java-ers. 

I will also move interface / class to its own file. Thanks for bringing this up 👍

Reply:
https://github.com/ably/kafka-connect-ably/pull/50/commits/58a941d99963649e0a38a360bb23b7a0e7e19f8d